oracle_fdw.c

```diff
@@ -821,9 +822,63 @@ oracleGetForeignRelSize(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntable
 				fdwState->oraTable->cols[i]->used = 1;
 	}
 
-	/* construct Oracle query and get the list of parameters and actions for RestrictInfos */
-	fdwState->query = createQuery(fdwState, baserel, for_update, root->query_pathkeys);
-	elog(DEBUG1, "oracle_fdw: remote query is: %s", fdwState->query);
+	/* createQuery is moved to GetForeignPlan. */
+	/* fdwState->for_update is needed by createQuery in GetForeignPlans */
+	fdwState->for_update = for_update;
```

Comment:
It is unnecessary to calculate `for_update` here and store it in `fdwState.`
Instead, it should be calculated in `oracleGetForeignPlan`, which is the only place where it is used.